azure-iot-device/azure/iot/device/common/pipeline/pipeline_stages_base.py

```diff
@@ -581,6 +585,26 @@ def _handle_pipeline_event(self, event):
                         self.name, event.name, event.request_id
                     )
                 )
+
+        elif isinstance(event, pipeline_events_base.ConnectedEvent):
+            """
+            If we're reconnecting, send all pending requests down again.  This is necessary
+            because any response that might have been sent by the service was likely lost
+            when the connection dropped.  The fact that the operation is still pending means
+            that we haven't received the response yet, and it's safe to re-send the request
+            because everything we do with this stage is allowed to repeat.  For instance, if
+            we update a twin twice with the same reported properties, the service will not
+            fail, likewise for getting a twin, or registering with the provisioning service.
+            """
+
+            self.send_event_up(event)
+
+            for request_id in self.pending_responses:
+                logger.info(
+                    "{}: ConnectedEvent: re-publishing request {}".format(self.name, request_id)
+                )
+                self._send_request_down(request_id, self.pending_responses[request_id])
```

Comment:
>_send_request_down [](start = 21, length = 18)

oh i think u wanted to reuse this method here.....that is why u did the change above...unless i am wrong..then plz explain to me..

Reply:
You are correct.

---
In reply to: [400450446](https://github.com/Azure/azure-iot-sdk-python/pull/520#discussion_r400450446) [](ancestors = 400450446)